pkg/printers/humanreadable.go

```diff
@@ -509,14 +509,17 @@ func (h *HumanReadablePrinter) PrintTable(obj runtime.Object, options PrintOptio
 		return nil, results[1].Interface().(error)
 	}
 
-	columns := handler.columnDefinitions
-	if !options.Wide {
-		columns = make([]metav1beta1.TableColumnDefinition, 0, len(handler.columnDefinitions))
-		for i := range handler.columnDefinitions {
-			if handler.columnDefinitions[i].Priority != 0 {
-				continue
+	var columns []metav1beta1.TableColumnDefinition
+	if !options.NoHeaders {
+		columns = handler.columnDefinitions
```

Comment:
we don't need the column definitions for anything non-display-related?

Reply:
In the code path that results in a table being created, no.  If you somehow always wired this to true and sent it to the server, `kubectl` would not correctly handle priority.

However, for all non-watch cases getting no headers is assumed you know what you are asking for (tabular data without headers), while in watch cases we define the behavior explicitly.